### ISSUE TITLE
`andX()` and `orX()` in WHERE clause not supported?

### ISSUE THREAD

**Issue by ThomasLandauer** [phpstan/phpstan-doctrine#415]
I'm building a query as shown at https://stackoverflow.com/a/16046356/1668200
```php
$queryBuilder = $this->createQueryBuilder('e');
$andX = $queryBuilder->expr()->andX();
$andX->add(...);
$orX = $queryBuilder->expr()->orX();
foreach (...) {
    $orX->add(...);
}
$andX->add($orX);
$queryBuilder
    ->where($andX)
    ->addOrderBy('e.foo', 'ASC')
;
return $queryBuilder->getQuery()->getArrayResult();
```
PHPStan is reporting:
> QueryBuilder: [Syntax Error] line 0, col 92: Error: Expected Literal, got 'ORDER'                                                                                
         DQL: SELECT ... FROM ... e WHERE  ORDER BY e.foo ASC

To me this looks like the PHPStan-internal(?) lexer doesn't support `andX()` and therefore creates an empty WHERE clause.

**Comment by ThomasLandauer** [comment]
Looks like this was fixed meanwhile, so I'm closing.
